FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90

```diff
@@ -335,6 +337,9 @@ subroutine atmosphere_init (Time_init, Time, Time_step, Grid_box, area)
       if (grids_on_this_pe(n)) mytile = n
    enddo
 
+   call coarse_graining_init(Atm(mytile)%flagstruct%npx, Atm(mytile)%npz, &
+        Atm(mytile)%layout, Atm(mytile)%bd, Atm(mytile)%coarse_graining)
+
```

Comment:
I don't quite understand why line 435 and others are wrapped with a 'if do_coarse_graining' but this one is not? I see those are for the diagnostics, but should there be a call to coarse_graining_init if np coarse graining is happening?

Reply:
Thanks @rheacangeo!  That's a good question.  The reason it's set up that way right now, is that `do_coarse_graining` is set in the `coarse_graining_nml`, which is only read in within the `coarse_graining_init` subroutine.  It might make more sense to set that particular namelist parameter at a higher level, say in `fv_core_nml`.  Do you think I should go with that?

Reply:
I went ahead with this change.  It is consistent with how the nudging module is initialized for instance: https://github.com/VulcanClimateModeling/fv3gfs-fortran/blob/31fc2ee360d9d5d3900cfb7b66393eabf8f05557/FV3/atmos_cubed_sphere/driver/fvGFS/atmosphere.F90#L444-L446.